City-of-Helsinki/drupal-helfi-platform-config
# UHF-3185: Clear React&Share localStorage items if cookie approval is not given

Clears React&Share localStorage items if user hasn't accepted statistics cookie category.

**How to test:**
* Get the changes: `composer require drupal/helfi_platform_config:dev-UHF-3185-clear-localstorage`.
* Clear caches: `drush cr`.
* Testing is easier if you exclude React and Share block from the Cookie settings page (`/cookie-information-and-settings`):
  * Go to `/admin/structure/block` and edit the  React and Share  block.
  * Exclude the page e.g. using the URL.
* Go to the `/cookie-information-and-settings` page and make sure statistics cookies are enabled.
* Next, go to a page which includes React and Share and interact with it.
* Using the browser's developer tools check that there is localStorage items `rnsbid`, `rnsbid_ts` and one or many that starts with `rns_reaction_`.
* Next, from the cookie settings page, disable statistics cookies.
* Now the localStorage items mentioned above should be gone.
* **Also check that this change is compatible with the newer React&Share change.**
